### ISSUE TITLE
security encoders

### ISSUE THREAD

**Issue by flip111** [symfony/symfony-docs#5877]
http://symfony.com/doc/current/cookbook/security/custom_provider.html#modify-security-yml

```
    encoders:
        Acme\WebserviceUserBundle\Security\User\WebserviceUser: sha512
```

http://symfony.com/doc/current/book/security.html#c-encoding-the-user-s-password
```
    encoders:
        Symfony\Component\Security\Core\User\User:
            algorithm: bcrypt
            cost: 12
```

1. Seems a bit weird that in the first instance you put the algorihm directly after the class and in the second case it has it's own key. 
2. What's the purpose of `Symfony\Component\Security\Core\User\User`? The first tutorial says you should make your own user class.
3. Sha512 seems a bit outdated by now, perhaps the doc needs some updating?

**Comment by JHGitty** [comment]
IMO: "Acme\WebserviceUserBundle\" is an old example. Currently we should use something like "AppBundle" or "UserBundle".

**Comment by xabbuh** [comment]
Let me answer to all your questions separately:

1. If you just want to configure the the algorithm to be used, you can use this shortcut notation which saves you some writing. Of course, that shortcut notation does not work when you have to configure more options of the encoder.
1. The cookbook you refer to is exactly about creating your own user provider while the book chapter just describes the basic stuff. It shows how to load users from memory and when you use that provider, your user's will be `Symfony\Component\Security\Core\User\User` objects.
1. We could update the example to use bcrypt there too.

And I agree we can also update the cookbook to use AppBundle examples (though that's not only this cookbook article, there are lots more using older examples).

**Comment by JHGitty** [comment]
This issue has beend fixed in #5890 , right?
